### PR TITLE
Objects dict instead of reward_locs

### PR DESCRIPTION
**Pull request by adamimos**
objects is an attribute of an environment that has up to two keys "rewards" and "markers"; each holds a dict describing the location and value. 

For graph_envs there is only a rewards keys, for grid_envs both are required, even if markers is empty.

### CODE REVIEW

**Comment by adamimos** [comment]
I've changed grid_env such that you can use either reward_locs OR objects, at least while using the reset function

File: neuronav/envs/grid_env.py [hunk 1]
```diff
@@ -158,10 +159,22 @@ def reset(
         else:
             self.agent_pos = self.agent_start_pos
 
+        # if both objects and reward_locs are None, throw an error
+        if objects == None and reward_locs == None:
```
Comment (awjuliani) [inline neuronav/envs/grid_env.py:163]: This should actually be fine, as in this case the behavior should be to do: `self.objects = self.topo_objects`

File: neuronav/envs/grid_env.py [hunk 2]
```diff
@@ -290,6 +313,11 @@ def make_visual_obs(self, resize=False):
         return img
 
     def make_window(self, w_size=2, block_size=20, resize=True):
+        """
```
Comment (awjuliani) [inline neuronav/envs/grid_env.py:316]: Thanks for adding this docstring!

File: neuronav/envs/grid_env.py [hunk 3]
```diff
@@ -58,7 +58,7 @@ def __init__(
             raise Exception("No valid OrientationType provided.")
         self.state_size *= self.orient_size
         self.agent_pos = [0, 0]
-        self.reward_locs = {}
+        self.objects = {} # a dict with keys of "reward" and "markers", each a dict
```
Comment (awjuliani) [inline neuronav/envs/grid_env.py:61]: Can you structure this as `self.objects = {'markers':{}, 'rewards':{}} ?

File: neuronav/envs/graph_env.py [hunk 4]
```diff
@@ -43,15 +43,15 @@ def __init__(
             self.images = utils.cifar10()
 
     def generate_graph(self, structure: GraphStructure):
-        self.rewarding_states, self.edges = structure_map[structure]()
+        self.objects, self.edges = structure_map[structure]()
+        self.rewarding_states = self.objects["rewards"] # TODO: is this redundant?
```
Comment (awjuliani) [inline neuronav/envs/graph_env.py:47]: I think doing similarly to `GrivEnv` would be best. Have a `self.struct_objects` that is used to reinitialize the `objects` in the case that none are passed in the reset.

**Review by awjuliani — COMMENTED** [review]
Overall looks great. There are a few minor changes I would suggest. Once those are taken care of, I'll be happy to merge this!

File: neuronav/envs/graph_env.py (hunk 4 above)
Comment (adamimos) [reply neuronav/envs/graph_env.py:47]: Ok I think I have fixed this

File: neuronav/envs/grid_env.py (hunk 3 above)
Comment (adamimos) [reply neuronav/envs/grid_env.py:61]: done

**Review by awjuliani — APPROVED** [review]
Looks great! Thanks again for contributing to the repo.

**Pull request merged**